tests/sidetrail/README.md

```diff
@@ -77,29 +115,102 @@ SideTrail proceeds in several steps:
+
+We will assume that you have a variable `$S2N` which encodes the path to s2n on your machine
 
-- bam-bam-boogieman 
-- SMACK and all its dependencies
-  - The easiest way to get these is to use the build.sh in smack/bin
-  - Ensure that all of the installed dependencies are on the $PATH
-  - source the smack.environment that the smack build script creates
+```shell
+export S2N=<path_to_s2n>
+```
+
+### Building the image
 
-### Execute the test
+To build the image, start with `s2n/codebuild/spec/sidetrail/Dockerfile`
 
+```shell
+cd $S2N
+docker build -f codebuild/spec/sidetrail/Dockerfile --tag sidetrail2 .
 ```
-cd tests/sidetrail/working/s2n-cbc
-./clean.sh
-./run.sh
 
-If the tests pass, you will see a message like: Boogie program verifier finished with 1 verified, 0 error
+This step takes about 25 minutes on my laptop.
 
-If the tests fail, you will see a message like: Boogie program verifier finished with 0 verified, 1 error
+### Starting docker
+
+```shell
+cd $S2N
+docker run -it -v `pwd`:/home/s2n --entrypoint /bin/bash sidetrail
 ```
 
+You will now be presented with a docker shell.
+Inside this shell, run:
+
+```shell
+source /sidetrail-install-dir/smack.environment 
+```
+
+This step is important.
+If you do not source this file when you begin working, SideTrail may appear to run, but not actually analyze the code.
+
+### Running a proof inside docker
+
+
+```shell
+cd /home/s2n/tests/sidetrail/working
+cd <testname>
+./clean.sh ; ./run.sh
+```
+
+You should see output that looks something like this 
+
+```
+...
+...
+warning: memory intrinsic length exceeds threshold (0); adding quantifiers.
+SMACK generated s2n_record_parse_wrapper@s2n_record_read_wrapper.c.compiled.bpl
+warning: module contains undefined functions: malloc, __CONTRACT_invariant, nondet
+
+ ____                _            _   
+|  _ \ _ __ ___   __| |_   _  ___| |_ 
+| |_) | '__/ _ \ / _` | | | |/ __| __|
+|  __/| | | (_) | (_| | |_| | (__| |_ 
+|_|   |_|  \___/ \__,_|\__,_|\___|\__|
+                                      
+s2n_record_parse_wrapper@s2n_record_read_wrapper.c
+
+
+
+__     __        _  __       
+\ \   / /__ _ __(_)/ _|_   _ 
+ \ \ / / _ \ '__| | |_| | | |
+  \ V /  __/ |  | |  _| |_| |
+   \_/ \___|_|  |_|_|  \__, |
+                       |___/ 
+s2n_record_parse_wrapper@s2n_record_read_wrapper.c
+
++  boogie /printModel 4 /doModSetAnalysis s2n_record_parse_wrapper@s2n_record_read_wrapper.c.product.bpl
+Boogie program verifier version 2.3.0.61016, Copyright (c) 2003-2014, Microsoft.
+
+Boogie program verifier finished with 1 verified, 0 errors
+
+real	0m23.334s
+user	0m20.983s
+sys	0m2.395s
+```
+
+If you do not see the line, 
+
+```
+Boogie program verifier finished with 1 verified, 0 errors
+```
+
+then you probably forgot to source `smack.environment`.
+Go back and do so.
+
+## Debugging SideTrail failures
 
+Consult [the debugging guide](DEBUGGING.md).
 ## Questions?
-contact dsn@amazon.com
\ No newline at end of file
+contact dsn@amazon.com
```

Comment:
Do we want to point to a team email alias instead? This would result in a single point of failure, when you are OOTO :)